csv.h

```diff
@@ -753,30 +787,26 @@ namespace io{
                 }
         };
 
-        template<char sep>
-        struct no_quote_escape{
-                static const char*find_next_column_end(const char*col_begin){
-                        while(*col_begin != sep && *col_begin != '\0')
-                                ++col_begin;
-                        return col_begin;
-                }
-
-                static void unescape(char*&, char*&){
+        
 
-                }
-        };
-
-        template<char sep, char quote>
+        template<char sep, char quote, bool null_terminated = true>
         struct double_quote_escape{
+				static bool hasQuote() {
+					return true;
+				}
+				static char getQuoteChar() {
+					return quote;
+				}
+
                 static const char*find_next_column_end(const char*col_begin){
-                        while(*col_begin != sep && *col_begin != '\0')
+                        while(*col_begin != sep && ( *col_begin != '\0'))
                                 if(*col_begin != quote)
                                         ++col_begin;
                                 else{
                                         do{
                                                 ++col_begin;
                                                 while(*col_begin != quote){
-                                                        if(*col_begin == '\0')
+                                                        if(( null_terminated && *col_begin == '\0'))
```

Comment:
if null_terminated template flag is false, encountering a '\0' inside a quoted column is not considered an escaped_string_not_closed error.
There is a use-case where quoted columns can have binary data inside them.